djangoevents/domain.py

```diff
@@ -47,8 +71,43 @@ class BaseEntity(BaseAggregate):
 
     OBSOLETE! Interface kept for backward compatibility.
     """
+    @classmethod
+    def is_abstract_class(cls):
+        return super().is_abstract_class() or cls is BaseEntity
 
     @classmethod
     def mutate(cls, entity=None, event=None):
         warnings.warn("`BaseEntity` is depreciated. Please switch to: `BaseAggregate`", DeprecationWarning)
         return super().mutate(aggregate=entity, event=event)
+
+
+def _list_subclasses(cls):
+    """
+    Recursively lists all subclasses of `cls`.
+    """
+    return cls.__subclasses__() + [g for s in cls.__subclasses__() for g in _list_subclasses(s)]
+
+
+def _list_internal_classes(cls, base_class=None):
+    base_class = base_class or object
+
+    return [cls_attribute for cls_attribute in cls.__dict__.values()
+            if inspect.isclass(cls_attribute)
+            and issubclass(cls_attribute, base_class)]
+
+
+def list_concrete_aggregates():
+    """
+    Lists all non abstract aggregates defined within the application.
+    """
+    aggregates = set(_list_subclasses(BaseAggregate) + _list_subclasses(BaseEntity))
+    return [aggregate for aggregate in list(aggregates) if not aggregate.is_abstract_class()]
+
+
+def list_aggregate_events(aggregate_cls):
+    """
+    Lists all aggregate_cls events defined within the application.
+    Note: Only events with a defined `mutate_event` flow will be returned.
+    """
```

Comment:
Note: I'm thinking about non-mutating events (like in `SampleEntity.Closed` example: https://github.com/ApplauseOSS/djangoevents/blob/cfcfb470f10d39f75dec008a26543bb0b63b55cb/djangoevents/tests/test_domain.py#L18). We should either list it here, change the name to list_aggregate_mutating_events or include a param (like `mutating_only=True`).